### PR TITLE
[Scala] setup Ivy cache directory

### PR DESCRIPTION
**Pull request by wing328**
Setup Ivy cache directory to `/home/circleci/.ivy2` to avoid download the dependencies in every run.

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/config/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

FYI @clasnake (2017/07), @jimschubert (2017/09) ❤️, @shijinkui (2018/01), @ramzimaalej (2018/03), @chameleon82 (2020/03), @Bouillie (2020/04)

### CODE REVIEW

File: samples/client/petstore/scala-httpclient/pom.xml [hunk 1]
```diff
@@ -157,6 +157,28 @@
...
+                        <id>sbt-test</id>
+                        <phase>integration-test</phase>
+                        <goals>
+                            <goal>exec</goal>
+                        </goals>
+                        <configuration>
+                            <executable>sbt</executable>
+                            <arguments>
+                                <argument>-ivy</argument>
+                                <argument>/home/circleci/.ivy2</argument>
```
Comment (jimschubert) [inline samples/client/petstore/scala-httpclient/pom.xml:175]: This won't fail when we run samples locally? I think these could be parameterized as a Maven property as well.

Comment (wing328) [reply samples/client/petstore/scala-httpclient/pom.xml:175]: Good point. Let me try `~` instead. I remember it won't expand ~ to home directory properly but let's give it another try.

Comment (wing328) [reply samples/client/petstore/scala-httpclient/pom.xml]: Done via https://github.com/OpenAPITools/openapi-generator/pull/6978/commits/aa5fcd8b4a93526c8b5245d3edde79fb42e59f01. Seems to work.

Comment (wing328) [reply samples/client/petstore/scala-httpclient/pom.xml]: UPDATE: it turns out `~` was not resolved properly. I've filed https://github.com/OpenAPITools/openapi-generator/pull/6985 to use `${user.home}` instead

**Pull request merged**